src/include/processor/operator/persistent/index_builder.h

```diff
@@ -25,14 +25,21 @@ namespace processor {
 
 const size_t SHOULD_FLUSH_QUEUE_SIZE = 32;
 
+constexpr size_t BUFFER_SIZE = 64;
+using OptionalExtraDataBuffer = std::optional<common::StaticVector<WarningSourceData, BUFFER_SIZE>>;
```

Comment:
This structure is quite large (2.5kB) so the `IndexBuilder` takes up quite a bit of memory (~660kB, all of which is on the stack during construction) so want to double-check that this won't cause any stack-usage related issues (I ran into some during this PR)

Reply:
Updated to allocate `IndexBuilder` on the heap, let me know if we want to resolve this some other way (e.g. making `IndexBuilder` smaller) and we can discuss